mobile/android/android-device.ts

```diff
@@ -185,9 +188,152 @@ export class AndroidDevice implements Mobile.IDevice {
+			    this.$childProcess.exec(uninstallCommand).wait();
+
+        var installCommand = this.composeCommand("install -r \"%s\"", packageFile);
+        this.$childProcess.exec(installCommand).wait();
+
+        var port = options["debug-port"];
+
+        var clearDebugEnvironmentCommand = this.composeCommand('shell rm "/sdcard/Android/data/%s/files/envDebug.out"', packageName);
+        this.$childProcess.exec(clearDebugEnvironmentCommand).wait();
+
+        this.startPackageOnDevice(packageName).wait();
+
+        var dbgPort = this.startAndGetPort(packageName).wait();
+
+        if (dbgPort > 0) {
+            this.tcpForward(dbgPort, dbgPort);
+            this.startDebuggerClient(dbgPort).wait();
+            this.openDebuggerClient(this.defaultNodeInspectorUrl + "?port=" + dbgPort).wait();
+        }
+    }
+
+    public debug(packageFile: string, packageName: string): IFuture<void> {
+        return (() => {
+            if (options["get-port"]) {
+                this.printDebugPort(packageName);
+            } else if (options["start"]) {
+                this.attachDebugger(packageName);
+            } else if (options["stop"]) {
+                this.detachDebugger(packageName);
+            } else if (options["debug-brk"]) {
+                this.startAppWithDebugger(packageFile, packageName);
+            } else {
+                this.$logger.info("Should specify at least one option: debug-brk, start, stop, get-port.");
+            }
+        }).future<void>()();
+    }
+
+    private checkIfRunning(packageName: string): boolean {
+        var isRunning = this.checkIfFileExists(packageName, "envDebug.out").wait();
+        return isRunning;
+    }
+
+    private checkIfFileExists(packageName: string, filename: string): IFuture<boolean> {
+        return (() => {
+            var args = ["shell", "test", "-f", "/sdcard/Android/data/" + packageName + "/files/" + filename, "&&", "echo 'yes'", "||", "echo 'no'"];
+            var res = this.$childProcess.spawnFromEvent(this.adb, args, "exit").wait();
+            var exists = res.stdout.indexOf('yes') > -1;
+            return exists;
+        }).future<boolean>()();
+    }
+
+    private startAndGetPort(packageName: string): IFuture<number> {
+        return (() => {
+            var port = -1;
+
+            var clearDebugEnvironmentCommand = this.composeCommand('shell rm "/sdcard/Android/data/%s/files/envDebug.in"', packageName);
+            this.$childProcess.exec(clearDebugEnvironmentCommand).wait();
+
+            var isRunning = false;
+            for (var i = 0; i < 10; i++) {
+                helpers.sleep(1000 /* ms */);
+                isRunning = this.checkIfRunning(packageName);
+                if (isRunning)
+                    break;
+            }
+
+            if (isRunning) {
+                var setEnvironmentCommand = this.composeCommand("shell \"cat /dev/null > \"/sdcard/Android/data/%s/files/envDebug.in\"\"", packageName);
```

Comment:
You can extract path "/sdcard/Android/data/%s/files/" as constant/variable, and use it is all references of this path.